app/Foliage/Meta.hs

```diff
@@ -121,10 +123,25 @@ data PackageVersionSpec = PackageVersionSpec
   deriving (Show, Eq, Generic)
   deriving anyclass (Binary, Hashable, NFData)
 
+matchMaybe :: forall a. Toml.TomlBiMap a Toml.AnyValue -> Toml.Key -> TomlCodec (Maybe a)
+matchMaybe bimap@(Toml.BiMap {}) key = Toml.Codec input output
+  where
+    input :: Toml.TomlEnv (Maybe a)
+    input toml = case HashMap.lookup key (Toml.tomlPairs toml) of
+      Nothing -> pure Nothing
+      Just anyVal -> pure <$> Toml.whenLeftBiMapError key (Toml.backward bimap anyVal) pure
+
+    output :: Maybe a -> Toml.TomlState (Maybe a)
+    output Nothing = pure Nothing
+    output (Just a) = do
+      anyVal <- Toml.eitherToTomlState $ Toml.forward bimap a
+      Just a <$ modify (Toml.insertKeyAnyVal key anyVal)
+
 sourceMetaCodec :: TomlCodec PackageVersionSpec
 sourceMetaCodec =
   PackageVersionSpec
-    <$> Toml.dioptional (timeCodec "timestamp")
+    -- TODO: unsure how to initialize Toml.BiMap ... with Toml._ZonedTime?
```

Comment:
I'm blocked here, otherwise I think this PR is mostly ready to be reviewed :)

Reply:
I pushed a commit to fill the gap.